### ISSUE TITLE
Collection<Relationship> is sometimes treated like a Relationship

### ISSUE THREAD

**Issue by cybersam** [neo4j/neo4j#4583]
This was observed using version 2.2.1.

**This query fails as expected**, since r1 is of type Collection<Relationship>:

```
MATCH (:User {name:"User"})-[r:IS_MEMBER_OF]->(:Group:Local)-[r1:IS_SUBGROUP_OF*0..]->(:Group)
Return type(r), TYPE(r1);
```

and the error message is:

```
> QueryExecutionKernelException: Type mismatch: expected Relationship but was Collection<Relationship> (line 1, column 121 (offset: 120))
==> "MATCH (:User {name:"User"})-[r:IS_MEMBER_OF]->(:Group:Local)-[r1:IS_SUBGROUP_OF*0..]->(:Group)     Return type(r), TYPE(r1)"
==>                                                                                                                          ^
```

**However, this query fails unexpectedly**:

```
MATCH (:User {name:"User"})-[r:IS_MEMBER_OF]->(:Group:Local)-[r1:IS_SUBGROUP_OF*0..]->(:Group)
Return type(r), EXTRACT (rel IN r1 | type(rel)) AS ancestorGroupTypes
```

with the following error message, which contradicts the assertion in the prior error message:

```
==> QueryExecutionKernelException: Expected `r1` to be a Collection<Any> but it was a Relationship
```

Also, the latter query does not cause an error message in the console (http://console.neo4j.org/?id=dcnna6).

**Comment by jexp** [comment]
Relationship identifiers in variable length paths are indeed holding a collection of relationships unlike single step relationships, that is intentional (but I agree it's confusing).

**Comment by jefersonmanjos** [comment]
I'm having the same mistake in this release. If I try to do any operation on the Collection <releationship>. For example, last (Collection <releationship>) and I get the error message

==> QueryExecutionKernelException: Expected `rs` to be a Collection<Any> but it was a Relationship

Any tips on how to solve it?

**Comment by lutovich** [comment]
Hello gentlemen,

I believe this issue was fixed by https://github.com/neo4j/neo4j/pull/4445.
So it would be released with 2.2.2 in a while and was released with 2.3.0-M01.
It would be great if you could check given queries agains 2.3.0-M01.

**Comment by cybersam** [comment]
I can no longer reproduce the problem using 2.3.0-M01, so it seems to be fixed.

**Comment by cosimosimeoneTR** [comment]
Hi there all!
Seems this bug in not fixed in final 2.3 EE version...

```
MATCH (in)-[rel*1..3]->(out) RETURN in,out,rel limit 9
```

is ok, while

```
MATCH (in)-[rel*1..3]->(out) RETURN in,out,type(rel) limit 9
```

gives syntax error on type(rel)

**Comment by jexp** [comment]
It is not a bug, in the var-length case rel is a collection of relationships.

Try this:

```
MATCH (in)-[rel*1..3]->(out) RETURN in,out,[r in rel | type(r)] limit 9
```
